src/components/ManaCalculator/index.tsx

```diff
@@ -0,0 +1,734 @@
+
+    delegatorResults = {
+      manaGeneratedPerEpoch: manaGeneratedPerEpoch,
+      passiveRewards: passiveRewards,
+      totalTPS: totalTPS,
+    };
+
+    manaGeneratedPerEpoch = calculateManaRewards(
+      state.stake,
+      state.delegator.validator,
+      {
+        performanceFactor: state.validator.performanceFactor,
+        fixedCost: state.validator.fixedCost,
+        shareOfYourStakeLocked: state.validator.shareOfYourStakeLocked,
+        attractedNewDelegatedStake: state.validator.attractedNewDelegatedStake,
+        attractedDelegatedStakeFromOtherPools:
+          state.validator.attractedDelegatedStakeFromOtherPools,
+      } as ValidatorParameters,
+      state.validators,
+      EPOCH,
+      null,
+      'Validator',
+    );
+    grantedTPS = calculateTPS(manaGeneratedPerEpoch, state.congestion);
+    totalTPS = grantedTPS + additionalTPS;
+
+    validatorResults.manaGeneratedPerEpoch = manaGeneratedPerEpoch;
+    validatorResults.passiveRewards = passiveRewards;
+    validatorResults.totalTPS = totalTPS;
+  }
+
+  // Calulate Mana rewards for delegator and validator
+  let delegatorAccumulateResults = {
+    manaGenerated: 0,
+    passiveRewards: 0,
+    totalTPS: 0,
+  };
+  let validatorAccumulateResults = {
+    manaGenerated: 0,
+    passiveRewards: 0,
+    totalTPS: 0,
+  };
+  {
+    let manaGenerated = calculateManaRewards(
+      state.stake,
+      state.delegator.validator,
+      null,
+      state.validators,
+      state.initialEpoch,
+      state.finalEpoch,
+      'Delegator',
+    );
+    let passiveRewards = calculatePassiveRewards(
+      state.stake,
+      state.initialEpoch,
+      state.finalEpoch,
+    );
+
+    let grantedTPS = calculateTPS(manaGenerated, state.congestion);
+    let additionalTPS = calculateTPS(passiveRewards, state.congestion);
+    let totalTPS = grantedTPS + additionalTPS;
+
+    delegatorAccumulateResults.manaGenerated = manaGenerated;
+    delegatorAccumulateResults.passiveRewards = passiveRewards;
+    delegatorAccumulateResults.totalTPS = totalTPS;
+
+    manaGenerated = calculateManaRewards(
+      state.stake,
+      state.delegator.validator,
+      {
+        performanceFactor: state.validator.performanceFactor,
+        fixedCost: state.validator.fixedCost,
+        shareOfYourStakeLocked: state.validator.shareOfYourStakeLocked,
+        attractedNewDelegatedStake: state.validator.attractedNewDelegatedStake,
+        attractedDelegatedStakeFromOtherPools:
+          state.validator.attractedDelegatedStakeFromOtherPools,
+      } as ValidatorParameters,
+      state.validators,
+      state.initialEpoch,
+      state.finalEpoch,
+      'Validator',
+    );
+    grantedTPS = calculateTPS(manaGenerated, state.congestion);
+    totalTPS = grantedTPS + additionalTPS;
+
+    validatorAccumulateResults.manaGenerated = manaGenerated;
+    validatorAccumulateResults.passiveRewards = passiveRewards;
+    validatorAccumulateResults.totalTPS = totalTPS;
+  }
```

Comment:
All this code is unnecessary and not used, it is the same you have in https://github.com/iotaledger/iota-wiki/blob/6fa6a7a5a6d6c5011e7903d63809fa6162eaf3ce/src/components/ManaCalculator/index.tsx#L264, it can be removed, right?